sessions/2018/december/Session_20.md

```diff
@@ -0,0 +1,955 @@
+@@ -28,6 +28,7 @@ public class CalculatorController {
+     }
+
+     @GetMapping("/factorial/{number}")
++    // JSON that contains a biginteger
+     public BigInteger getFactorial(@PathVariable("number") int number){
+         return calculator.getFactorial(number);
+     }
+diff --git a/src/test/java/co/org/osso/api/ApiApplicationCalculatorTests.java b/src/test/java/co/org/osso/api/ApiApplicationCalculatorTests.java
+index fa172d1..a0fdd36 100644
+--- a/src/test/java/co/org/osso/api/ApiApplicationCalculatorTests.java
++++ b/src/test/java/co/org/osso/api/ApiApplicationCalculatorTests.java
+@@ -1,16 +1,41 @@
+ package co.org.osso.api;
+
++import com.fasterxml.jackson.core.JsonProcessingException;
++import com.fasterxml.jackson.databind.DeserializationFeature;
++import com.fasterxml.jackson.databind.ObjectMapper;
++import org.apache.http.HttpResponse;
++import org.apache.http.auth.AuthScope;
++import org.apache.http.auth.UsernamePasswordCredentials;
++import org.apache.http.client.CredentialsProvider;
++import org.apache.http.client.methods.HttpPost;
++import org.apache.http.client.methods.HttpUriRequest;
++import org.apache.http.entity.StringEntity;
++import org.apache.http.impl.client.BasicCredentialsProvider;
++import org.apache.http.impl.client.CloseableHttpClient;
++import org.apache.http.impl.client.HttpClientBuilder;
++import org.apache.http.util.EntityUtils;
+```
+
+And then we added:
+
+```bash
+[jaar@port-staff api]$ git add . ; git commit -m "Add HTTP verbs Integration test"; git push
+[http-integration-tests c6fe812] Add HTTP verbs Integration test
+ 2 files changed, 84 insertions(+)
+Enumerating objects: 31, done.
+Counting objects: 100% (31/31), done.
+Delta compression using up to 4 threads
+Compressing objects: 100% (8/8), done.
+Writing objects: 100% (17/17), 2.53 KiB | 863.00 KiB/s, done.
+Total 17 (delta 3), reused 0 (delta 0)
+remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
+To github.com:javarb/api.git
+   3e3245d..c6fe812  http-integration-tests -> http-integration-tests
+```
+
+Fetch to sync local master with remote master and tree of remote braches locally:
+
+```bash
+[jaar@port-staff api]$ git fetch
+remote: Enumerating objects: 2, done.
+remote: Counting objects: 100% (2/2), done.
+remote: Compressing objects: 100% (2/2), done.
+remote: Total 2 (delta 0), reused 0 (delta 0), pack-reused 0
+Unpacking objects: 100% (2/2), done.
+From github.com:javarb/api
+   f77c479..37c4a2d  master     -> origin/master
+
+[jaar@port-staff api]$ git branch -a
+  api-basics
+  api-calculator
+  api-calculator-fix-bug
+  bugsnag-integration
+* http-integration-tests
+  master
+  remotes/origin/api-basics
+  remotes/origin/api-calculator
+  remotes/origin/api-calculator-fix-bug
+  remotes/origin/bugsnag-integration
+  remotes/origin/http-integration-tests
+  remotes/origin/master
+```
+
+And we add the `-p` flag to sync deleting to avoid to see locally, remote branchs that doesn't existed in really remotely:
```

Comment:
Could you please note down that this `-p` flag has nothing to do with the `git add -p` flag, and that this one stands for `prune`, whereas the `git add -p` one stands for `partial`.